### PR TITLE
ftp: tests to show some firewall issues - v1

### PR DESCRIPTION
**Pull request by jasonish**
Adds some baseline tests then some firewall tests.

Shows had ftpdata_command: retr is not possible to match against in firewall
model as ftpdata_command makes some assumptions on direction that don't always
hold up.

### CODE REVIEW

File: tests/firewall/ruletype-firewall-112-ftp-download-passive/test.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      count: 1
+      match:
+        event_type: ftp_data
+        ftp_data.command: RETR
+
+  # ftpdata_command: retr matches in ftp-data:response_started
+  - filter:
+      count: 1
+      match:
+        alert.signature_id: 5000
```
Comment (victorjulien) [inline tests/firewall/ruletype-firewall-112-ftp-download-passive/test.yaml:35]: check for 5010 is missing, think it would be good to add for completeness

**Comment by victorjulien** [comment]
The firewall tests both are positive tests: matching the rules, allowing the traffic. Can you also add tests for negative rulesets (explicit drop, or drop by rule omission). In addition, I think it would be good to test the default policy side as well.

**Comment by victorjulien** [comment]
Not expecting much difference, but both fw tests use `retr`, would be good to complete it by adding 2 `stor` based tests too?

File: tests/firewall/ruletype-firewall-112-ftp-download-passive/test.yaml (hunk 1 above)
Comment (jasonish) [reply tests/firewall/ruletype-firewall-112-ftp-download-passive/test.yaml:35]: Added, makese sense.

**Comment by jasonish** [comment]
> The firewall tests both are positive tests: matching the rules, allowing the traffic. Can you also add tests for negative rulesets (explicit drop, or drop by rule omission). In addition, I think it would be good to test the default policy side as well.

Those are in another branch; this was more specific to show the issues in https://github.com/OISF/suricata/pull/15656. But it would be nice to clear this out first.

**Comment by jasonish** [comment]
> Not expecting much difference, but both fw tests use `retr`, would be good to complete it by adding 2 `stor` based tests too?

Make sense.

**Comment by jasonish** [comment]
> Not expecting much difference, but both fw tests use `retr`, would be good to complete it by adding 2 `stor` based tests too?

Added, just mirror the download tests for now.

**Comment by jasonish** [comment]
> > The firewall tests both are positive tests: matching the rules, allowing the traffic. Can you also add tests for negative rulesets (explicit drop, or drop by rule omission). In addition, I think it would be good to test the default policy side as well.
> 
> Those are in another branch; this was more specific to show the issues in [OISF/suricata#15656](https://github.com/OISF/suricata/pull/15656). But it would be nice to clear this out first.

Mainly because other fixes are required to do something like allow certain commands and disallow others.

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #3170, thanks!
